qcodes/plots/qcmatplotlib.py

```diff
@@ -434,42 +436,34 @@ def _draw_pcolormesh(self, ax, z, x=None, y=None, subplot=1,
 
         pc = ax.pcolormesh(*args, **kwargs)
 
+        # Scale colors from clim kwarg, or otherwise from min(z) and max(z)
+        data_arrays = [data_array.get_array() for data_array in ax.collections]
+        if clim is None:
+            # Get color limits as min/max of all existing plotted 2D arrays
+            clim = [np.min([np.nanmin(data_array) for data_array in data_arrays]),
+                    np.max([np.nanmax(data_array) for data_array in data_arrays])]
+
+        # Update color limits for all plotted 2D arrays
+        for mesh in ax.collections:
```

Comment:
Does this raise issues if you have 1D data as well as 2D data?

Reply:
@nulinspiratie I think we can merge this too today.

Reply:
We might've been able to, but alas. Let's see if we can still do it this week, I'll have a look

Reply:
@maij Not sure if I simply missed these messages in my inbox or if I never received them, sorry

Reply:
Had a look, ax.collections does not contain 1D data. @maij I think it's ready to be merged